src/main/java/uk/gov/hmcts/ccd/data/casedetails/search/builder/GrantTypeSqlQueryBuilder.java

```diff
@@ -146,4 +149,28 @@ public String getOperator(String query, String operator) {
         }
         return EMPTY;
     }
+
+    private String addInQueryForCaseAccessCategory(CaseTypeDefinition caseType,
+                                                   SearchRoleAssignment representative,
+                                                   String parentQuery) {
+        String caseAccessCategoriesQuery = getCaseAccessCategoriesQuery(caseType, representative);
+        if (StringUtils.isNotBlank(caseAccessCategoriesQuery)) {
+            parentQuery = parentQuery + getOperator(parentQuery, AND)
+                + String.format(QUERY_WRAPPER, caseAccessCategoriesQuery);
+        }
+        return parentQuery;
+    }
+
+    private String getCaseAccessCategoriesQuery(CaseTypeDefinition caseType,
+                                                SearchRoleAssignment representative) {
+        List<String> caseAccessCategories = caseType.getRoleToAccessProfiles().stream()
+            .filter(rap -> rap.getRoleName().equalsIgnoreCase(representative.getRoleName()))
+            .filter(rap -> rap.getCaseAccessCategories() != null)
```

Comment:
If there is at least one mapped access profile that has no category, then we don't require adding the clause at all. I don't think we're covering that point at the moment?